overlord/registrystate/registrystate.go

```diff
@@ -442,3 +520,49 @@ func GetStoredTransaction(t *state.Task) (*Transaction, *state.Task, error) {
 func setTransaction(t *state.Task, tx *Transaction) {
 	t.Set("registry-transaction", tx)
 }
+
+func IsRegistryHook(ctx *hookstate.Context) bool {
+	return !ctx.IsEphemeral() &&
+		(strings.HasPrefix(ctx.HookName(), "change-view-") ||
+			strings.HasPrefix(ctx.HookName(), "save-view-") ||
+			strings.HasSuffix(ctx.HookName(), "-view-changed"))
+}
+
+// Context is used by GetTransaction to defer actions until a point after the
+// call (e.g., blocking on a registry transaction only when the caller wants to).
+// The context also carries a hookstate.Context that is used to determine whether
+// the registry is being accessed from a snap hook.
+type Context struct {
+	hookCtx *hookstate.Context
+
+	onDoneCallbacks []func() error
+}
+
+func NewContext(ctx *hookstate.Context) *Context {
+	regCtx := &Context{
+		hookCtx: ctx,
+	}
+
+	if ctx != nil {
+		ctx.OnDone(func() error {
+			return regCtx.Done()
+		})
+	}
+
+	return regCtx
+}
+
+func (c *Context) onDone(f func() error) {
+	c.onDoneCallbacks = append(c.onDoneCallbacks, f)
+}
+
+func (c *Context) Done() error {
```

Comment:
why do we export this?

Reply:
Because the caller (which include daemon/) will have to call it to trigger the new change